[beta.31] ng e2e does not work anymore

### OS?
Archlinux

### Versions.
> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:

```
@angular/cli: 1.0.0-beta.31
node: 7.5.0
os: linux x64
@angular/common: 2.4.6
@angular/compiler: 2.4.6
@angular/core: 2.4.6
@angular/flex-layout: 2.0.0-beta.4
@angular/forms: 2.4.6
@angular/http: 2.4.6
@angular/material: 2.0.0-beta.1
@angular/platform-browser: 2.4.6
@angular/platform-browser-dynamic: 2.4.6
@angular/router: 3.4.6
@angular/cli: 1.0.0-beta.31
@angular/compiler-cli: 2.4.6
```

### Repro steps.

Created with ng init and always updated

Running `ng e2e --progress=false -e=prod-e2e` gives:
```

➜  frontend git:(master) ✗ ng e2e --progress=false -e=prod-e2e                               
** NG Live Development Server is running on http://localhost:4200. **
(node:24803) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
Hash: af5145659c048450397c
Time: 16784ms
chunk    {0} 0.chunk.js, 0.bundle.map 190 kB {1} {2} {3} {4} {5} {6} {7} {8} {10} [rendered]
chunk    {1} 1.chunk.js, 1.bundle.map 157 kB {0} {2} {3} {4} {5} {6} {7} {8} {10} [rendered]
chunk    {2} 2.chunk.js, 2.bundle.map 111 kB {0} {1} {3} {4} {5} {6} {7} {8} {10} [rendered]
chunk    {3} 3.chunk.js, 3.bundle.map 18.3 kB {0} {1} {2} {4} {5} {6} {7} {8} {10} [rendered]
chunk    {4} 4.chunk.js, 4.bundle.map 19.8 kB {0} {1} {2} {3} {5} {6} {7} {8} {10} [rendered]
chunk    {5} 5.chunk.js, 5.bundle.map 18.9 kB {0} {1} {2} {3} {4} {6} {7} {8} {10} [rendered]
chunk    {6} 6.chunk.js, 6.bundle.map 17 kB {0} {1} {2} {3} {4} {5} {7} {8} {10} [rendered]
chunk    {7} 7.chunk.js, 7.bundle.map 26.6 kB {0} {1} {2} {3} {4} {5} {6} {8} {10} [rendered]
chunk    {8} 8.chunk.js, 8.bundle.map 11.5 kB {0} {1} {2} {3} {4} {5} {6} {7} {10} [rendered]
chunk    {9} polyfills.bundle.js, polyfills.bundle.map (polyfills) 222 kB {14} [initial] [rendered]
chunk   {10} main.bundle.js, main.bundle.map (main) 202 kB {13} [initial] [rendered]
chunk   {11} styles.bundle.js, styles.bundle.map (styles) 203 kB {14} [initial] [rendered]
chunk   {12} scripts.bundle.js, scripts.bundle.map (scripts) 15.3 kB {14} [initial] [rendered]
chunk   {13} vendor.bundle.js, vendor.bundle.map (vendor) 4.14 MB [initial] [rendered]
chunk   {14} inline.bundle.js, inline.bundle.map (inline) 0 bytes [entry] [rendered]

WARNING in ./src/app/features/cockpit/workspaces/petals-content/petals-component-view/petals-component-overview/petals-component-overview.component.ts
35:55-68 "export 'IComponentRow' was not found in '../../../state/components/component.interface'

WARNING in ./src/app/features/cockpit/workspaces/petals-content/petals-component-view/petals-component-overview/petals-component-overview.component.ts
35:88-101 "export 'IComponentRow' was not found in '../../../state/components/component.interface'

WARNING in ./src/app/features/cockpit/workspaces/petals-content/petals-service-unit-view/petals-service-unit-overview/petals-service-unit-overview.component.ts
35:55-70 "export 'IServiceUnitRow' was not found in '../../../state/service-units/service-unit.interface'

WARNING in ./src/app/features/cockpit/workspaces/petals-content/petals-service-unit-view/petals-service-unit-overview/petals-service-unit-overview.component.ts
35:90-105 "export 'IServiceUnitRow' was not found in '../../../state/service-units/service-unit.interface'
webpack: Compiled with warnings.
(node:24803) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module 'protractor/node_modules/webdriver-manager/built/lib/cmds/update' from '/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend'
(node:24803) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

And stay blocked.

And running directly `ng e2e --progress=false -e=prod-e2e --serve=false` gives:

```
(node:25037) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
Cannot find module 'protractor/node_modules/webdriver-manager/built/lib/cmds/update' from '/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend'
Error: Cannot find module 'protractor/node_modules/webdriver-manager/built/lib/cmds/update' from '/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend'
    at Function.module.exports [as sync] (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/resolve/lib/sync.js:36:11)
    at Object.requireProjectModule (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/utilities/require-project-module.js:5:28)
    at /home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/tasks/e2e.js:28:66
    at Class.run (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/tasks/e2e.js:9:16)
    at Class.run (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/commands/e2e.js:43:28)
    at Class.<anonymous> (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/ember-cli/lib/models/command.js:134:17)
    at process._tickCallback (internal/process/next_tick.js:103:7)

```

@filipesilva :
```
➜  frontend git:(master) ✗ yarn list protractor
yarn list v0.20.0
└─ protractor@5.1.1
```

Before it was at version 5.1.0 and I manually updated it, so I went back to 5.1.0 and retried:
```
➜  frontend git:(master) ✗ ng e2e --progress=false -e=prod-e2e --serve=false
(node:26056) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
Cannot find module 'protractor/node_modules/webdriver-manager/built/lib/cmds/update' from '/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend'
Error: Cannot find module 'protractor/node_modules/webdriver-manager/built/lib/cmds/update' from '/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend'
    at Function.module.exports [as sync] (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/resolve/lib/sync.js:36:11)
    at Object.requireProjectModule (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/utilities/require-project-module.js:5:28)
    at /home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/tasks/e2e.js:28:66
    at Class.run (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/tasks/e2e.js:9:16)
    at Class.run (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/commands/e2e.js:43:28)
    at Class.<anonymous> (/home/vnoel/Linagora/Petals/dev/git/petals-cockpit-new/frontend/node_modules/@angular/cli/ember-cli/lib/models/command.js:134:17)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

--- a/packages/@angular/cli/tasks/e2e.ts
+++ b/packages/@angular/cli/tasks/e2e.ts
@@ -1,4 +1,5 @@
 import * as url from 'url';
+import { stripIndents } from 'common-tags';
 
 import { E2eTaskOptions } from '../commands/e2e';
 import { CliConfig } from '../models/config';
@@ -38,10 +39,25 @@ export const E2eTask = Task.extend({
       }
 
       if (e2eTaskOptions.webdriverUpdate) {
-        // webdriver-manager can only be accessed via a deep import from within
-        // protractor/node_modules. A double deep import if you will.
-        const webdriverUpdate = requireProjectModule(projectRoot,
-          'protractor/node_modules/webdriver-manager/built/lib/cmds/update');
+        // The webdriver-manager update command can only be accessed via a deep import.
+        const webdriverDeepImport = 'webdriver-manager/built/lib/cmds/update';
+        let webdriverUpdate: any;
+
+        try {
+          // When using npm, webdriver is within protractor/node_modules.
+          webdriverUpdate = requireProjectModule(projectRoot,
+            `protractor/node_modules/${webdriverDeepImport}`);
+        } catch (e) {
+          try {
+            // When using yarn, webdriver is found as a root module.
+            webdriverUpdate = requireProjectModule(projectRoot, webdriverDeepImport);
+          } catch (e) {
+            throw new SilentError(stripIndents`
+              Cannot automatically find webdriver-manager to update.
+              Update webdriver-manager manually and run 'ng e2e --no-webdriver-update' instead.
+            `);
+          }
+        }
         // run `webdriver-manager update --standalone false --gecko false --quiet`
         // if you change this, update the command comment in prev line, and in `eject` task
         promise = promise.then(() => webdriverUpdate.program.run({

